app/src/main/java/com/github/sdp_begreen/begreen/adapters/ExpandableListAdapter.kt

```diff
@@ -0,0 +1,129 @@
+package com.github.sdp_begreen.begreen.adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseExpandableListAdapter
+import android.widget.ImageView
+import android.widget.TextView
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.models.Group
+import com.github.sdp_begreen.begreen.models.Item
+
+// This adapter is responsible for displaying expandable lists with groups and their respective items.
+class ExpandableListAdapter(
+    private val context: Context,
+    private val groups: List<Group>
+) : BaseExpandableListAdapter() {
+
+    // Returns the number of groups in the list.
+    override fun getGroupCount(): Int {
+        return groups.size
+    }
+
+    // Returns the number of children (items) in a particular group.
+    override fun getChildrenCount(groupPosition: Int): Int {
+        if (groupPosition < 0 || groupPosition >= groups.size) {
+            throw IllegalArgumentException()
+        }
+        return groups[groupPosition].items.size
+    }
+
+
+    // Returns the group object at the specified position.
+    override fun getGroup(groupPosition: Int): Any {
+        if (groupPosition < 0 || groupPosition >= groups.size) {
+            throw IllegalArgumentException()
+        }
+        return groups[groupPosition]
+    }
+
+    // Returns the child (item) object in the specified group and child position.
+    override fun getChild(groupPosition: Int, childPosition: Int): Any {
+        if (groupPosition < 0 || groupPosition >= groups.size) {
+            throw IllegalArgumentException()
+        }
+        return groups[groupPosition].items[childPosition]
+    }
+
+    // Returns a unique identifier for a group.
+    override fun getGroupId(groupPosition: Int): Long {
+        if (groupPosition < 0 || groupPosition >= groups.size) {
+            throw IllegalArgumentException()
+        }
+        return groupPosition.toLong()
+    }
+
+    // Returns a unique identifier for a child (item) within a group.
+    override fun getChildId(groupPosition: Int, childPosition: Int): Long {
+        if (groupPosition < 0 || groupPosition >= groups.size) {
```

Comment:
You should also check that `childPosition` lie in the correct range

Reply:
That's right! It is changed, thanks!